codebuild/spec/buildspec_tsan.yml

```diff
@@ -0,0 +1,25 @@
+---
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not use
+# this file except in compliance with the License. A copy of the License is
+# located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+# implied. See the License for the specific language governing permissions and
+# limitations under the License.
+version: 0.2
+
+phases:
+  build:
+    on-failure: ABORT
+    commands:
+      - cmake . -Bbuild -DTSAN=on
+      - cmake --build ./build -j $(nproc)
+  post_build:
+    on-failure: ABORT
+    commands:
+      - CTEST_OUTPUT_ON_FAILURE=1 CTEST_PARALLEL_LEVEL=$(nproc) make -C build test
```

Comment:
nit: Would it make sense to add this to the environment variables that you set in the CMakeLists.txt? That way all of our CI would get it for free. It would make it hard to suppress failed test output, but that doesn't seem like something that would ever be necessary.

Reply:
Oh that's a good idea! Then we don't have to remember it everywhere. I'll do that in a separate PR.